crates/node_runtime/src/node_runtime.rs

```diff
@@ -215,29 +223,36 @@ impl NodeRuntime for RealNodeRuntime {
             }
 
             let mut command = Command::new(node_binary);
-            command.env_clear();
-            command.env("PATH", env_path);
+            command
+                .env_clear()
+                .env("PATH", env_path)
+                .arg(npm_file)
+                .arg(subcommand)
+                .args(["--cache".into(), installation_path.join("cache")])
+                .args([
+                    "--userconfig".into(),
+                    installation_path.join("blank_user_npmrc"),
+                ])
+                .args([
+                    "--globalconfig".into(),
+                    installation_path.join("blank_global_npmrc"),
+                ])
+                .args(args);
+
             // Set proxy arg, if user's env has `http_proxy`.
             if let Some(http_proxy) = std::env::var("http_proxy").ok() {
                 command.env("http_proxy", http_proxy);
             }
-            command.arg(npm_file).arg(subcommand);
-            command.args(["--cache".into(), installation_path.join("cache")]);
-            command.args([
-                "--userconfig".into(),
-                installation_path.join("blank_user_npmrc"),
-            ]);
-            command.args([
-                "--globalconfig".into(),
-                installation_path.join("blank_global_npmrc"),
-            ]);
-            command.args(args);
 
             if let Some(directory) = directory {
-                command.current_dir(directory);
-                command.args(["--prefix".into(), directory.to_path_buf()]);
+                command
+                    .current_dir(directory)
+                    .args(["--prefix".into(), directory.to_path_buf()]);
             }
 
+            #[cfg(windows)]
+            command.creation_flags(windows::Win32::System::Threading::CREATE_NO_WINDOW.0);
```

Comment:
@maxdeviant I'd like to improve the `Command::new` in entire of the Zed, there have a lot of place need to do this.

> For example: `crates\languages\src\go.rs`

So, can I add a helper in `util::process:command` in `util` crate to create `smol::process:Command` in shortly with Windows special `creation_flags`? Then other place can just use this method to create command.

> Oh, I found that the comment was in the wrong place. I should post in PR #11156